src/Surfnet/StepupGateway/GatewayBundle/Entity/SamlEntity.php

```diff
@@ -110,6 +109,16 @@ public function toServiceProvider()
         $configuration['entityId']             = $this->entityId;
         $configuration['configuredLoas']       = $decodedConfiguration['loa'];
 
+        $configuration['secondFactorOnly'] = false;
+        if (isset($decodedConfiguration['second_factor_only'])) {
+            $configuration['secondFactorOnly'] = (bool) $decodedConfiguration['second_factor_only'];
```

Comment:
The type of `$decodedConfiguration['second_factor_only']` is validated in Middleware, right? If this has failed, I'd rather not ignore the issue by casting it to a boolean (eg. (`(bool) "false" === true`).